bluesky/callbacks/zmqsub.py

```diff
@@ -0,0 +1,102 @@
+import multiprocessing
+import ast
+import zmq
+import zmq.asyncio
+import asyncio
+import time
+from ..utils import expiring_function
+from ..run_engine import Dispatcher, DocumentNames
+
+
+_loop = zmq.asyncio.ZMQEventLoop()
+asyncio.set_event_loop(_loop)
```

Comment:
This was the most confusing aspect of asyncio + zmq. From my experiments this morning, I think the correct (ZMQ-supporting) loop has to be set up at coroutine *definition* time. I can't think of a good reason why this should be true, but it does seem to be true.